### PR TITLE
Warn user when trying to move or delete selected features outside of map view

### PR DESCRIPTION
**Pull request by wonder-sk**
Sometimes users may inadvertently move/delete features they are not aware of being selected. If that is the case, extra warning message box will be shown for confirmation.

Does not apply for deletion from attribute table - it is assumed that features do not need to be visible when deleting from attribute table.

In QgisApp::deleteSelected() method the last argument was not used anymore, so I have replaced it by the "checkFeaturesVisible" argument.

Developed for Arpa Piemonte (Dipartimento Tematico Geologia e Dissesto) within ERIKUS project.

### CODE REVIEW

File: src/app/qgisapp.cpp [hunk 1]
```diff
@@ -7140,9 +7140,31 @@ void QgisApp::deleteSelected( QgsMapLayer *layer, QWidget *parent, bool promptCo
     return;
   }
   //display a warning
-  if ( promptConfirmation && QMessageBox::warning( parent, tr( "Delete features" ), tr( "Delete %n feature(s)?", "number of features to delete", numberOfSelectedFeatures ), QMessageBox::Ok | QMessageBox::Cancel ) == QMessageBox::Cancel )
+  if ( checkFeaturesVisible )
   {
-    return;
+    QgsFeature feat;
+    QgsFeatureIterator it = vlayer->getSelectedFeatures( QgsFeatureRequest().setSubsetOfAttributes( QgsAttributeList() ) );
```
Comment (nyalldawson) [inline src/app/qgisapp.cpp]: I think you'll need to also add a destination matching the canvas here

**Comment by nyalldawson** [comment]
Thanks - I've certainty fallen into this trap myself a lot!

File: src/app/qgisapp.cpp (hunk 1 above)
Comment (wonder-sk) [reply src/app/qgisapp.cpp]: What did you mean, to add destination CRS to the feature request? two lines below the canvas view rectangle is reprojected to layer CRS...

Comment (nyalldawson) [reply src/app/qgisapp.cpp]: Oops... Missed that!

**Comment by nyalldawson** [comment]
@wonder-sk is this going to land for 3.0? I think it should - it's an important fix.

**Comment by wonder-sk** [comment]
Ah yes, I have just forgotten to merge it... Will fix the conflict and merge.

**Pull request merged**